### ISSUE TITLE
Timestamp timezones not preserved for certain indexers

### ISSUE THREAD

**Issue by azjps** [pandas-dev/pandas#12938]
Hi, using different indexers on a dataframes with timezone-aware timestamps may return naive timestamps. (A brief search attempt didn't turn up any related issues.)
#### Code Sample, a copy-pastable example if possible

``` python
>>> import pandas as pd
>>> df = pd.Series([pd.tslib.Timestamp('2016-03-30 14:35:25.506088-04:00')]).to_frame()
>>> df = pd.concat([df, df]).reset_index(drop=True)
>>> df[0][0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')
>>> df.iloc[0,0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')
>>> df.loc[0,0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')
>>> df.iat[0,0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')
>>> df.at[0,0]
Timestamp('2016-03-30 18:35:25.506088')
>>> df.ix[0,0]
Timestamp('2016-03-30 18:35:25.506088')
>>> df.ix[0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')
>>> df[0].at[0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')
>>> df[0].ix[0]
Timestamp('2016-03-30 14:35:25.506088-0400', tz='pytz.FixedOffset(-240)')

```
#### Expected Output

I'd expect `df.at[0,0]` and `df.ix[0,0]` to show the same timezone-aware timestamps as the other indexing methods.
#### output of `pd.show_versions()`

``` python
>>> import pandas as pd
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.18.12-el6.ia32e.lime.0
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0
nose: None
pip: 7.1.0
setuptools: 19.4
Cython: None
numpy: 1.11.0
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.5.2
pytz: 2016.3
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
```

**Comment by jreback** [comment]
yeah this is probably not exhaustively tested. patches welcome.

**Comment by jreback** [comment]
closing as duplicate of #15822

**Comment by jorisvandenbossche** [comment]
Running the above example again with latest master:

```
In [60]: df = pd.Series([pd.Timestamp('2016-03-30 14:35:25', tz='Europe/Brussels')]).to_frame()

In [61]: df = pd.concat([df, df]).reset_index(drop=True)

In [62]: df
Out[62]: 
                          0
0 2016-03-30 14:35:25+02:00
1 2016-03-30 14:35:25+02:00

In [63]: df[0][0]
Out[63]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [64]: df.iloc[0,0]
Out[64]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [65]: df.loc[0,0]
Out[65]: Timestamp('2016-03-30 12:35:25')

In [66]: df.iat[0,0]
Out[66]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [67]: df.at[0,0]
Out[67]: Timestamp('2016-03-30 12:35:25')

In [68]: df[0].loc[0]
Out[68]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [69]: df[0].at[0]
Out[69]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')
```

So the `df.loc[0,0]` case has regressed to also not return the timezone.

**Comment by jreback** [comment]
@jorisvandenbossche I think that last is noted in the other issue. If you (or others!) want to try to pin down *when* this changed would be great (otherwise) I will fix soon.

**Comment by jreback** [comment]
```
In [5]: df[0][0]
Out[5]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [6]: df.iloc[0,0]
Out[6]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [7]:  df.loc[0,0]
Out[7]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [8]: df.iat[0,0]
Out[8]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [9]:  df.at[0,0]
Out[9]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [10]: df[0].loc[0]
Out[10]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')

In [11]: df[0].at[0]
    ...: 
Out[11]: Timestamp('2016-03-30 14:35:25+0200', tz='Europe/Brussels')
```

after #15827

**Comment by jreback** [comment]
tests: https://github.com/pandas-dev/pandas/commit/66fb0a3e0c25cf10988ce0d14a7efee437aa94d6
